source/common/access_log/access_log_formatter.cc

```diff
@@ -299,5 +331,52 @@ std::string RequestHeaderFormatter::format(const Http::HeaderMap& request_header
   return HeaderFormatter::format(request_headers);
 }
 
+MetadataFormatter::MetadataFormatter(const std::string& filter_namespace,
+                                     const std::vector<std::string>& path,
+                                     const absl::optional<size_t>& max_length)
+    : filter_namespace_(filter_namespace), path_(path), max_length_(max_length) {}
+
+std::string MetadataFormatter::format(const envoy::api::v2::core::Metadata& metadata) const {
+  const auto filter_it = metadata.filter_metadata().find(filter_namespace_);
+  if (filter_it == metadata.filter_metadata().end()) {
+    return UnspecifiedValueString;
+  }
+  const ProtobufWkt::Struct* dataStruct = &(filter_it->second);
+  const Protobuf::Message* data = dataStruct;
+  // go through path to select sub entries
+  for (const auto p : path_) {
```

Comment:
Can you refactor this lookup part of the method to be a general lookup helper in https://github.com/envoyproxy/envoy/blob/master/source/common/config/metadata.h?